lib/puppet/property.rb

```diff
@@ -102,6 +102,22 @@ def array_matching=(value)
       raise ArgumentError, "Supported values for Property#array_matching are 'first' and 'all'" unless [:first, :all].include?(value)
       @array_matching = value
     end
+
+    # Used to mark a type property as having or lacking idempotency (on purpose
+    # generally). This is used to avoid marking the property as a
+    # corrective_change when there is known idempotency issues with the property
+    # rendering a corrective_change flag as useless.
+    # @return [Boolean] true if the property is marked as non-idempotent
```

Comment:
"Returns true if the property is non-idempotent" sounds backwards. Is that just a docs issue?